modules/Material/Dialog.qml

```diff
@@ -19,118 +19,186 @@
-
-	default property alias contents: mainCol.children
-	property string negativeBtnText: "CANCEL"
-	property string positiveBtnText: "OK"
-	property real minWidth: negativeBtn.width + positiveBtn.width + units.dp(124) //min 100dp padding
-	property real maxHeight: view.parent.height/2
-        property real minHeight: units.dp(96) + titleLabel.height
-	property string title
-
-	signal accepted()
-	signal rejected()
-
-	function open() {
-		visible = true;
-	}
-
-	function close() {
-		visible = false;
-	}
-
-	width: view.parent.width/2 >= minWidth ? view.parent.width/2 : minWidth
-	height: {
-        	if (units.dp(96) + mainCol.height <= maxHeight && units.dp(96) + mainCol.height >= minHeight) {
-            		return units.dp(96) + mainCol.height;
-        	} else if (units.dp(96) + mainCol.height < minHeight) {
-            		return minHeight;
-        	} else {
-            		return maxHeight;
-        	}
-    	}
-	elevation: 5
-	anchors.centerIn: parent
-
-	Flickable {
-        id: mainFlick
-
-		anchors {
-			top: parent.top
-			left: parent.left
-			bottom: btnContainer.top
-			right: parent.right
-			rightMargin: units.dp(24)
-			topMargin: units.dp(24)
-			leftMargin: units.dp(24)
-			bottomMargin: units.dp(16)
-		}
-		clip: true
-		interactive: contentHeight > height
-		contentHeight: mainCol.height
-        	contentWidth: width
-
-		Column {
-			id: mainCol
-
-           		spacing: units.dp(5)
-            		width: parent.width
-
-			Label {
-                		id: titleLabel
-
-				style: "title"
-                		width: parent.width
-				text: view.title
-                		wrapMode: Text.Wrap
-			}
-		}
-	}
-
-	Item {
-		id: btnContainer
-
-		width: parent.width
-		height: units.dp(48)
-		anchors.bottom: parent.bottom
-		anchors.bottomMargin: units.dp(8)
-
-		Button {
-			id: negativeBtn
-
-			text: view.negativeBtnText
-			anchors {
-				verticalCenter: parent.verticalCenter
-				right: positiveBtn.left
-				rightMargin: units.dp(8)
-			}
-			onClicked: {
-				view.close();
-				view.rejected();
-			}
-		}
-
-		Button {
-			id: positiveBtn
-
-			text: positiveBtnText
-			anchors {
-				verticalCenter: parent.verticalCenter
-				right: parent.right
-				rightMargin: units.dp(16)
-			}
-			onClicked: {
-				view.close();
-				view.accepted();
-			}
-		}
-	}
-
-	Scrollbar {
-        	flickableItem: mainFlick
-    	}
+import Material.Extras 0.1
+
+Popover {
+    id: dialog
+    anchor: Item.Center
+    backdrop: true
+
+    width: Math.max(dialogContainer.width, minimumWidth)
+    height: dialogContainer.height
+
+    property int minimumWidth: units.dp(240)
+
+    property alias title: titleLabel.text
+    property string negativeBtnText: "CANCEL"
+    property string positiveBtnText: "OK"
+    property bool hasActions: true
+
+    default property alias dialogContent: placeholder.children
+
+    signal accepted()
+    signal rejected()
+
+    transitions: [
+        Transition {
+            from: "open"
+            to: "closed"
```

Comment:
You may need to use `from: "*"` and `to: "*"`, which is what I've seen in examples. I'm not sure if you can just remove them completely.